blog/2022-08-19-jit-service account/index.md

```diff
@@ -0,0 +1,72 @@
+
+Below are the steps and modules of Paralus that are involved in provisioning of just in time service accounts
+
+1. Users log in to Paralus and request to execute a kubectl command on a target cluster to create a name server.
+2. The request passes on to **Relay Server** which in turn communicates with the **Paralus core** to validate the user's roles and permission.
+3. If the user has valid access permissions, Paralus core responds with a **yaml configuration file** with the credentials and provides it to the Relay server.
+4. The relay server passes on that configuration to the **relay agent** that is running on the target Kubernetes cluster.
+5. On the target cluster, the configuration is validated and a **service account is created** by Kubernetes.
+6. Once that is confirmed by the relay agent, the relay server then pushes the user's request to the relay agent.
+7. At the target cluster, the user request is once again validated with their permissions for the service account.
+8. If the user has the relevant permissions, the request is processed and a namespace is created.
+
+Things to remember about Just in time Service accounts on Paralus:
+
+- The default TTL for a service account is 5 minutes.
+- These accounts are deleted after 8 hours of inactivity from the target cluster.
+
+## Benefits of Just In Time Service Accounts
+
+### Short-lived Scoped Access
+
+The whole concept of Just in time access revolves around being **short-lived and with limited scope**. Paralus provides service accounts on target clusters for a specific duration (5 minutes) and with permissions that are assigned by the organization admin has been assigned to limit the scope.
+
+This significantly brings down the attack surface as users have access only to a specific scope. Further, since these just in time service accounts are short-lived, they prevent privileg theft too.
+
+### No Blanket Roles
+
+By default Kubernetes creates a default service account if you don't specify/configure one. All the commands are executed using this default service account. If every user in your organization uses the default service account, you are inviting trouble.
+
+You don't want any user to access any resource they want. Hence, Paralus provides a **just in time service account** creation for every user and avoids blanket roles.
+
+### Custom Role Creation
+
+When users in your organization grow, it becomes tedious to manage individual user roles. You might end up having to manage more roles than resources! To avoid that, Paralus allows you to **[create custom roles](../docs/usage/roles#creating-roles/)**.
```

Comment:
just wondering how custom roles solves the problem mentioned here. "When users in your organization grow, it becomes tedious to manage individual user roles." and also what relavance it has with JIT

Reply:
Makes sense, removed this entire section and added another one related to auditing and compliance.